codesearch/server/server.go

```diff
@@ -272,13 +315,24 @@ func (css *codesearchServer) Index(ctx context.Context, req *inpb.IndexRequest)
 
 		log.Infof("Starting indexing %q@%s", repoURL, commitSHA)
 
-		r, err := css.syncIndex(ctx, req)
-		if err != nil {
-			log.Errorf("Failed indexing %q: %s", repoURL, err)
-			return err
+		if req.GetReplacementStrategy() == inpb.ReplacementStrategy_INCREMENTAL {
+			r, err := css.incrementalUpdate(ctx, req)
+			if err != nil {
+				log.Errorf("Failed indexing %q: %s", req.GetGitRepo().GetRepoUrl(), err)
+				return err
+			}
+			rsp = r
+		} else if req.GetReplacementStrategy() == inpb.ReplacementStrategy_REPLACE_REPO {
+			r, err := css.fullyReindex(ctx, req)
+			if err != nil {
+				log.Errorf("Failed indexing %q: %s", req.GetGitRepo().GetRepoUrl(), err)
+				return err
+			}
+			rsp = r
+		} else {
+			return status.InvalidArgumentErrorf("Invalid replacement strategy %s", req.GetReplacementStrategy())
 		}
```

Comment:
Could do something like this 

```suggestion
		var err error
		if req.GetReplacementStrategy() == inpb.ReplacementStrategy_INCREMENTAL {
			rsp, err = css.incrementalUpdate(ctx, req)
		} else if req.GetReplacementStrategy() == inpb.ReplacementStrategy_REPLACE_REPO {
			rsp, err = css.fullyReindex(ctx, req)
		} else {
			return status.InvalidArgumentErrorf("Invalid replacement strategy %s", req.GetReplacementStrategy())
		}
		if err != nil {
			log.Errorf("Failed indexing %q: %s", req.GetGitRepo().GetRepoUrl(), err)
			return err
		}
```

Reply:
good call